### PR TITLE
API.md updates

### PR DESCRIPTION
**Pull request by tlisonbee**
- Adding secrets API, healthcheck endpoint, trigger job endpoint.
- Switching to X-Cerberus-Token
- Adding some example error responses
- Various clean-up

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -622,27 +652,117 @@ This endpoint allows a user to delete a safe deposit box that they own
...
+
+    + Body
+
+        {
+          "request_id" : "bbdd111111a2aaa3a",
+          "lease_id" : "",
+          "renewable" : false,
+          "lease_duration" : 3600,
+          "data" : {
+            "keys" : [ "path1", "path2", "path3" ]
```
Comment (fieldju) [inline API.md]: if a key ends with `/` ex: `path4/` then it is a virtual path and doesn't contain secure data.

Not sure if or how that should be incorporated into this example

File: API.md [hunk 2]
```diff
@@ -153,10 +165,14 @@ refresh_count and max_refresh_count can be used to determine when a re-authentic
 
 ### Authenticate with Cerberus as an App [POST]
 
-This endpoint takes IAM ARN information and generates an base 64 encoded KMS encrypted payload of the below. The ARN if registered with an SDB will have kms decrypt permissions on the KMS key that the payload was enrypted with.
+This endpoint takes IAM ARN information and generates a base64 encoded KMS encrypted payload.
```
Comment (fieldju) [inline API.md]: The response object from this request is as follows

    {
      "auth_data": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
    }

The value of "auth_data" is the kms encrypted base64 encoded payload that has a structure of 

    {
       "client_token" : "e5fd901b-bc65-71e7-a214-6066fef1e918",
       "policies" : [ "foo-bar-read", "lookup-self" ],
       "metadata" : {
         "aws_region" : "us-west-2",
         "iam_principal_arn" : "arn:aws:iam::111111111:role/fake-role"
         "username" : "arn:aws:iam::111111111:role/fake-role"
         "is_admin": "false",
         "groups": "registered-iam-principals"
       },
       "lease_duration" : 3600,
       "renewable" : true
     }

I feel like that could be documented better here and was always a point of confusion for people trying to make new clients or do things manually.

File: API.md [hunk 3]
```diff
@@ -121,7 +132,8 @@ refresh_count and max_refresh_count can be used to determine when a re-authentic
 
     + Headers
 
-            X-Vault-Token: 7f6808f1-ede3-2177-aa9d-45f507391310
+            X-Cerberus-Token: 7f6808f1-ede3-2177-aa9d-45f507391310
```
Comment (fieldju) [inline API.md]: Nit: it doesn't really matter but tokens no longer contain non-alphanumeric chars, no more hyphens.

Comment (tlisonbee) [reply API.md]: Updated

File: API.md (hunk 2 above)
Comment (tlisonbee) [reply API.md]: There were two examples that were different from each other.  I updated them to be the same and used the better example.

**Pull request merged**